flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/FlinkConfigOptions.java

```diff
@@ -94,7 +94,7 @@ private FlinkConfigOptions() {}
   public static final ConfigOption<SplitAssignerType> TABLE_EXEC_SPLIT_ASSIGNER_TYPE =
       ConfigOptions.key("table.exec.iceberg.split-assigner-type")
           .enumType(SplitAssignerType.class)
-          .defaultValue(SplitAssignerType.SIMPLE)
+          .defaultValue(null)
```

Comment:
Is this intended? This would be a breaking change if a user did not provide a split assigner type before

Reply:
good catch. 
I changed that to null, as we had an assertion before:
```
      if (splitAssignerFactory != null) {
        Preconditions.checkArgument(
            watermarkColumn == null,
            "Watermark column and SplitAssigner should not be set in the same source");
      }
```

however, after my work on this PR, it is safe to remove the above assertion, since we will override the factory with a `OrderedSplitAssignerFactory` when the `watermarkColumn` is specified

Reply:
Makes sense. It would be good to update the docs to reveal how the default is determined

Reply:
I'm updating the doc entry for `watermark-column`:

```
Specifies the watermark column to use for watermark generation. Setting this value will override the `splitAssignerFactory` option.
```
wdyt? @mas-chen @pvary

Reply:
Sounds good. If users have not opt-ed into the new feature, I would expect their SQL queries to still work. Can you add tests to ensure that?

Also, I think we want to set this to `.noDefaultValue`

Reply:
Not according to the code.
```
   if (watermarkColumn != null) {
        // Column statistics is needed for watermark generation
        context = context.copyWithColumnStat(watermarkColumn);
        SplitWatermarkExtractor watermarkExtractor =
            new ColumnStatsWatermarkExtractor(icebergSchema, watermarkColumn, watermarkTimeUnit);
        emitter = SerializableRecordEmitter.emitterWithWatermark(watermarkExtractor);
        splitAssignerFactory =
            new OrderedSplitAssignerFactory(SplitComparators.watermark(watermarkExtractor));
      }
```

Reply:
cc: @pvary

Reply:
If the splits are not ordered, then we will have fluctuating watermarks. We do not emit those, which are not in order, but beats the purpose of the whole watermark generation feature.
Imagine a situation where we reading time series data, and read the latest file first. Every other file will contain late data in this case, and might be dropped.

So while technically possible, I rather not allow the users to shoot themselves in the foot.

Reply:
> If users have not opt-ed into the new feature, I would expect their SQL queries to still work. Can you add tests to ensure that?

Hi! I didn't get your suggestion about the unit test. Would you please rephrase? Thanks @mas-chen